### ISSUE TITLE
Hyphens in API versions lead to incorrect CRDs

### ISSUE THREAD

**Issue by grdryn** [operator-framework/operator-sdk#2162]
## Bug Report

**What did you do?**
```
operator-sdk add api --api-version=my-cool-thing.grdryn.me/v1alpha1 --kind Test
```

**What did you expect to see?**
I expected the generated CRD to be at `deploy/crds/my-cool-thing_v1alpha1_test_crd.yaml`, and have the group set to `my-cool-thing.grdryn.me`.

**What did you see instead? Under which circumstances?**
It was generated as `deploy/crds/my-cool-thing_v1alpha1_test_crd.yaml` and the group in the file was de-hyphenated to:
```
mycoolthing.grdryn.me
```

I think this is incorrect, as in other places I can see the correct group specified, such as in `pkg/apis/mycoolthing/v1alpha1/register.go`. If I then create a controller for this, the generate CRD & CR yaml files aren't what the controller will react to.

**Environment**
* operator-sdk version:
```
operator-sdk version: v0.10.1, commit: 872e7d997486bb587660fc8d6226eaab8b5c1087
```

* go version:
```
go version go1.13.3 linux/amd64
```

* Are you writing your operator in ansible, helm, or go?
Go

**Possible Solution**

I debugged this a little, and think this might be the problem:
https://github.com/operator-framework/operator-sdk/blob/master/internal/scaffold/resource.go#L133

This is called by the openapi generation stuff, and throws away the hyphens.

Hyphens are maybe not a great idea anyway, so maybe the easiest thing to do is probably to just check for non-alphanumeric characters on the `operator-sdk add api [...]` command, and fail with a message if there are some there?

**Additional context**

I hit this with an operator that I inherited, where `deploy/crds/` has both the incorrectly generated CRDs like above, and some older manually created CRD yaml files that are correct (but different than the generated ones, and missing all validation, etc). The manually created ones are in production, but now I can't get the operator-sdk to update those correct ones with the default openapi generation.

**Comment by camilamacedo86** [comment]
HI @grdryn, 

The validation to ensure that the Group is a valid value for k8s API is done [here](https://github.com/operator-framework/operator-sdk/blob/master/internal/scaffold/resource.go#L135). 

The point that you raised [here](https://github.com/operator-framework/operator-sdk/blob/master/internal/scaffold/resource.go#L133) need to be replaced because it will set the value to the GroupImport which is used to generate the go imports. 

The places where the file names are defined are:
- CR: https://github.com/operator-framework/operator-sdk/blob/424a61d56000e6e3d91d352faa1bd4f7c814661f/internal/scaffold/cr.go#L51
- CRD: https://github.com/operator-framework/operator-sdk/blob/424a61d56000e6e3d91d352faa1bd4f7c814661f/internal/scaffold/crd.go#L71

Then, I agree with you that we have here 2 possible issues. The names for CR/CRD has not been generated with the same format and may the group var used is with "-" replaces when should not. 

I will be working on it. Thank you for raising this issue.

**Comment by camilamacedo86** [comment]
Also, it is related to https://github.com/operator-framework/operator-sdk/issues/2058

**Comment by estroz** [comment]
@grdryn this issue has come up a few times before (#1812, #997, #1368) and apparently has not been fixed despite those issues being closed. I will note that file naming is not the issue here; with the `spec.version` field being deprecated in CRD's in favor of `spec.versions`, we changed CRD file names to `<full group>_<resource>_crd.yaml`. More on that in our [version migration guide](https://github.com/operator-framework/operator-sdk/blob/master/doc/migration/version-upgrade-guide.md#operator-sdk-updates).

We will look into this further and hopefully resolve this hyphen/non-hyphen discrepancy once and for all.

**Comment by estroz** [comment]
@grdryn the hyphen problem was fixed in SDK v0.11.0. The issue related to how our CRD generator managed hyphens in group names. I suggest upgrading to `operator-sdk` v0.11.0+. Check out our upgrade guide for [v0.11.x](https://github.com/operator-framework/operator-sdk/blob/master/doc/migration/version-upgrade-guide.md#v011x) if you decide to upgrade SDK versions. If that is not possible then either:

1. Rename the CRD group name manually.
2. Rename the group name in `register.go` manually.

Feel free to re-open this issue if our comments didn't help resolve your problem. Thanks!

**Comment by camilamacedo86** [comment]
HI @grdryn,  

Just to clarifies and supplement the issue that you are reporting regards the group names is solved as @estroz described. It is a duplication of https://github.com/operator-framework/operator-sdk/issues/1812#issuecomment-533814837.  (solved in 0.11+)

The other point regards the version no longer be part of the CRD name is also something that was changed and it in the above link. Just to share, see that when we upgrade the version we still using the same CRD: https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/ so because of this no make sense keep the version on its name.

**Comment by grdryn** [comment]
Thanks folks, glad to see it was fixed in v0.11.0. I'll see if I can figure out a way to upgrade to that version (will need to find time to try a workaround like in https://github.com/operator-framework/operator-sdk/issues/2118#issuecomment-549598512 though).
